components/provider_cldr/src/transform/dates.rs

```diff
@@ -130,6 +130,21 @@ impl From<&cldr_json::StylePatterns> for gregory::patterns::StylePatternsV1 {
     }
 }
 
+impl From<&cldr_json::DateTimeFormats> for gregory::patterns::DateTimeFormatsV1 {
+    fn from(other: &cldr_json::DateTimeFormats) -> Self {
+        // TODO(#308): Support numbering system variations. We currently throw them away.
+        Self {
+            style_patterns: gregory::patterns::StylePatternsV1 {
+                full: other.full.get_pattern().clone(),
+                long: other.long.get_pattern().clone(),
+                medium: other.medium.get_pattern().clone(),
+                short: other.short.get_pattern().clone(),
+            },
+            available_formats: other.available_formats.0.clone(),
```

Comment:
You don't have to copy the available_formats literally from CLDR into ICU4X.  Is there any pre-processing you can do to optimize them?